### ISSUE TITLE
Model containing EfficientNet cannot be restored

### ISSUE THREAD

**Issue by asterisk37n** [tensorflow/tensorflow#43806]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS 10.15.6 / Colab
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: None
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below):v2.3.0-54-gfcc4b966f1 2.3.1 (Local),  v2.3.0-54-gfcc4b966f1 2.3.1 (Colab)
- Python version: 3.7.3 (Local), 3.6.9 (Colab)
- Bazel version (if compiling from source): None
- GCC/Compiler version (if compiling from source): None
- CUDA/cuDNN version: None
- GPU model and memory: None

**Describe the current behavior**
1. Trained a model containing pretrained EfficientNetB0 with GradientTape without compiling a model
2. Saved a model by ```model.save('path')```
3. Restoreing a model shows warning that gradient cannot be requested
4. Finetuning fails

**Describe the expected behavior**
Successfully save, restore and finetune.

**Standalone code to reproduce the issue**

[Colab](https://colab.research.google.com/drive/1gzOwSWJ1Kvwzo01SEpjqGq6Lb-OsI-ob?usp=sharing)
[StackOverflow](https://stackoverflow.com/questions/64218503/finetune-savedmodel-failure-due-to-no-gradient-loaded)

**Other info / logs** Include any logs or source code that would be helpful to
I followed this tutorial. How can I save and restore to train again with different learning rate for finetuning?
https://www.tensorflow.org/tutorials/quickstart/advanced

At first I posted on StackOverflow and then tried to reproduce on Colab to give more information, but error messages are different. It might be because of data loading difference, but I'm not sure.
Thank you for reading.

**Comment by asterisk37n** [comment]
This happens when I use EfficientnetB0, not happens when I use ResNet50
https://colab.research.google.com/drive/1ea4lXkZHs3QksRDAXKJzTFZa-lWvcPJL?usp=sharing

**Comment by asterisk37n** [comment]
I found this third party repository is a workaround
https://github.com/qubvel/efficientnet

**Comment by bhack** [comment]
Can you unlock colab access?

**Comment by asterisk37n** [comment]
@bhack I unlocked. Thank you for checking.

**Comment by bhack** [comment]
Check https://github.com/tensorflow/tensorflow/issues/43576

**Comment by asterisk37n** [comment]
@bhack 
Thank you. It is a related issue. It seems no solution for now. So, If this is not a bug, is this expected to happen? Thank you!

**Comment by bhack** [comment]
Yes probably you can close this and subscribe to that one or https://github.com/tensorflow/tensorflow/issues/40166
